### DISCUSSION TITLE
idf.py menuconfig

### DISCUSSION THREAD

**Discussion by soky** [Mair/esp32-course#152] [General, feedback and testimonials]
I am having two problems with idf.py  1) flash  2) menuconfig   they result in errors i.e   cmake failed with exit code 1.
I already installed esp-idf when starting Mair's Class.  Here is the complete error code list for menuconfig error:

Idf.py menuconfig

C:\Espressif\frameworks\esp-idf-v4.4>idf.py menuconfig
Executing action: menuconfig
Running cmake in directory c:\espressif\frameworks\esp-idf-v4.4\build
Executing "cmake -G Ninja -DPYTHON_DEPS_CHECKED=1 -DESP_PLATFORM=1 -DCCACHE_ENABLE=1 c:\espressif\frameworks\esp-idf-v4.4"...
-- The C compiler identification is unknown
-- The CXX compiler identification is unknown
-- The ASM compiler identification is unknown
-- Didn't find assembler
CMake Error at CMakeLists.txt:2 (project):
  No CMAKE_C_COMPILER could be found.

  Tell CMake where to find the compiler by setting either the environment
  variable "CC" or the CMake cache entry CMAKE_C_COMPILER to the full path to
  the compiler, or to the compiler name if it is in the PATH.
CMake Error at CMakeLists.txt:2 (project):
  No CMAKE_CXX_COMPILER could be found.
  Tell CMake where to find the compiler by setting either the environment
  variable "CXX" or the CMake cache entry CMAKE_CXX_COMPILER to the full path
  to the compiler, or to the compiler name if it is in the PATH.

CMake Error at CMakeLists.txt:2 (project):
  No CMAKE_ASM_COMPILER could be found.
  Tell CMake where to find the compiler by setting either the environment
  variable "ASM" or the CMake cache entry CMAKE_ASM_COMPILER to the full path
  to the compiler, or to the compiler name if it is in the PATH.
-- Warning: Did not find file Compiler/-ASM
-- Configuring incomplete, errors occurred!
See also "C:/Espressif/frameworks/esp-idf-v4.4/build/CMakeFiles/CMakeOutput.log".
See also "C:/Espressif/frameworks/esp-idf-v4.4/build/CMakeFiles/CMakeError.log".
cmake failed with exit code 1
C:\Espressif\frameworks\esp-idf-v4.4>

I would gratefully appreciate any help.  If need I will post the idf.py flash error full listing as well.

**Comment by Mair** [comment]
You are in the wrong directory. You are running it from  your idf installation but you need to run it from your project

**Comment by soky** [comment]
I cd'd to the directory I wanted to use. Perhaps I didnt go deep enough.  

Thanks so much for your continued help Mair.

soky
